specification/dsdl/dsdl.tex

```diff
@@ -376,6 +431,108 @@ \section{Primitive data types}\label{sec:dsdl_primitive_data_types}
                             \\
 \end{UAVCANSimpleTable}
 
+\subsection{Primitive constant data types}\label{sec:dsdl_primitive_constant_data_types}
+
+In addition to the ordinary data types that are used for defining types to be transmitted over the wire,
+there also exists data types only used for constant evaluation.
+The const data types are used for things that can be calculated only with the information self contained in the dsdl,
+such as defining array lengths, assertions and constant definitions.
+All the primitive const types are given below together with the literal used for construction.
+
+\begin{UAVCANSimpleTable}{Primitive constant data types}{|l l l|}
+    \label{table:dsdl_constant_primitive_data_types}
+    Name &
+        Legal Values &
+        Literal (regex) \\
+
+    \footnotesize \texttt{const bool} &
+        \footnotesize $\{true, false\}$ &
+        \footnotesize \texttt{true | false} \\
+
+    \footnotesize \texttt{const int} &
+        \footnotesize $\supseteq \lbrack -\frac{2^{64}}{2}, 2^{64} - 1\rbrack$ &
+        \footnotesize \texttt{0 |
+                            (\textbackslash+|-)?[1-9][0-9]* |
+                            (\textbackslash+|-)?0x([0-9]|[a-f])+ |
+                            (\textbackslash+|-)?0o[0-7]+ |
+                            (\textbackslash+|-)?0b(1|2)+ } \\
+
+    \footnotesize \texttt{const float} &
+        \footnotesize \texttt{IEEE 754-2008 (binary64)} &
+        \footnotesize \texttt{(\textbackslash+|-)?[0-9]*.[0-9]* |
+                            \textbackslash+|-)?[0-9]*((e|E)-?[0-9]+) |
+                            \textbackslash+|-)?[0-9]*.[0-9]*((e|E)-?[0-9]+) } \\
+
+    \footnotesize \texttt{const char} &
+        \footnotesize $\lbrack0, 127\rbrack$ &
+        \footnotesize
+            \texttt{ '[\^{}'\textbackslash\textbackslash]' |
+                'x[0-9]+' |
```

Comment:
The conventional form is `'\\x[0-9]+'`, is it not true? The leading backslash seems to be missing?

Reply:
Thanks, willfix